src/test/regress/expected/update_gp.out

```diff
@@ -209,6 +208,418 @@ DROP TABLE keo2;
+               ->  Seq Scan on t1
+ Optimizer: Postgres query optimizer
+(9 rows)
+
+UPDATE t1 SET j = t_strewn.i FROM t_strewn WHERE t_strewn.i = t1.i
+RETURNING *;
+ i  | j  | i  
+----+----+----
+  9 |  9 |  9
+  6 |  6 |  6
+ 12 | 12 | 12
+  3 |  3 |  3
+(4 rows)
+
+EXPLAIN (costs off)
+WITH CTE AS (DELETE FROM t1 RETURNING *)
+SELECT count(*) AS a FROM t_strewn JOIN cte USING (i);
+                                QUERY PLAN                                 
+---------------------------------------------------------------------------
+ Aggregate
+   ->  Gather Motion 3:1  (slice2; segments: 3)
+         ->  Aggregate
+               ->  Hash Join
+                     Hash Cond: (t_strewn.i = cte.i)
+                     ->  Seq Scan on t_strewn
+                     ->  Hash
+                           ->  Broadcast Motion 3:3  (slice1; segments: 3)
+                                 ->  Subquery Scan on cte
+                                       ->  Delete on t1
+                                             ->  Seq Scan on t1
+ Optimizer: Postgres query optimizer
+(12 rows)
+
+WITH CTE AS (DELETE FROM t1 RETURNING *)
+SELECT count(*) AS a FROM t_strewn JOIN cte USING (i);
+ a 
+---
+ 4
+(1 row)
+
+EXPLAIN (costs off)
+DELETE FROM t_strewn WHERE t_strewn.i = (SELECT t2.i FROM t2 WHERE t_strewn.i = t2.i);
+                              QUERY PLAN                               
+-----------------------------------------------------------------------
+ Delete on t_strewn
+   ->  Seq Scan on t_strewn
+         Filter: (i = (SubPlan 1))
+         SubPlan 1  (slice0; segments: 3)
+           ->  Result
+                 Filter: (t_strewn.i = t2.i)
+                 ->  Materialize
+                       ->  Broadcast Motion 3:3  (slice1; segments: 3)
+                             ->  Seq Scan on t2
+ Optimizer: Postgres query optimizer
+(10 rows)
+
+DELETE FROM t_strewn WHERE t_strewn.i = (SELECT t2.i FROM t2 WHERE t_strewn.i = t2.i)
+RETURNING *;
+ i  
+----
+  3
+  9
+  6
+ 12
+(4 rows)
+
+EXPLAIN (costs off)
+UPDATE t_strewn SET i = t_strewn2.i
+FROM t_strewn2 WHERE t_strewn.i = t_strewn2.i;
+                               QUERY PLAN                               
+------------------------------------------------------------------------
+ Update on t_strewn
+   ->  Explicit Redistribute Motion 3:3  (slice3; segments: 3)
+         ->  Hash Join
+               Hash Cond: (t_strewn.i = t_strewn2.i)
+               ->  Redistribute Motion 3:3  (slice1; segments: 3)
+                     Hash Key: t_strewn.i
+                     ->  Seq Scan on t_strewn
+               ->  Hash
+                     ->  Redistribute Motion 3:3  (slice2; segments: 3)
+                           Hash Key: t_strewn2.i
+                           ->  Seq Scan on t_strewn2
+ Optimizer: Postgres query optimizer
+(12 rows)
+
+UPDATE t_strewn SET i = t_strewn2.i
+FROM t_strewn2 WHERE t_strewn.i = t_strewn2.i
+RETURNING *;
+ i  | i  
+----+----
+  5 |  5
```

Comment:
Why will the tuple order be always the same?

Reply:
Doesn't pg_regress diff ignore the order of these results? I think it only checks that the right results are present in any order.